### PR TITLE
Issue/260

### LINKED ISSUE #260: /register middleName field is disable by default
The TCK is always sending the middleName as field in registration even though the spec says it should be disable by default.

According to the spec:

```
middleName:
            enabled: false
            visible: true
            label: "Middle Name"
            placeholder: "Middle Name"
            required: true
            type: "text"
```

A list of fields, as defined by stormpath.web.register.form.fields, and ordered by stormpath.web.register.form.fieldOrder. Fields should only be in the list if their enabled property is true. As such the enabled property can be omitted from each list element. Any enabled fields not in fieldOrder should be appended to the end of the list.

And finally 

If the post contains a field that is disabled or not defined by the developer, the library MUST reject the request with an error. We do not allow arbitrary data to be posted to an account's custom data object.

### PR DESCRIPTION
**Pull request by jarias**
Fixes #260

### CODE REVIEW

**Comment by nbarbettini** [comment]
Looks good. Can we also add a test to verify that an error occurs if you try to POST with middleName or username?

**Comment by jarias** [comment]
@nbarbettini I added the extra test.

**Comment by dogeared** [comment]
bump @nbarbettini

File: src/test/groovy/com/stormpath/tck/register/RegisterIT.groovy [hunk 1]
```diff
@@ -176,6 +190,26 @@ class RegisterIT extends AbstractIT {
...
 
+    @Test(groups=["v100", "json"])
+    public void registerErrorsForDisableDefaultFields() throws Exception {
+
+        def jsonMap = [email: testAccount.email,
+                       password: testAccount.password,
+                       givenName: testAccount.givenName,
+                       surname: testAccount.surname,
+                       middleName: "foobar"]
+        // field 'customValue' is not defined in the default configuration
```
Comment (nbarbettini) [inline src/test/groovy/com/stormpath/tck/register/RegisterIT.groovy]: stray comment?

**Comment by nbarbettini** [comment]
Added one comment, otherwise LGTM 👍

**Pull request merged**